src/main/java/de/tum/in/www1/artemis/web/rest/BonusResource.java

```diff
@@ -0,0 +1,253 @@
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+/**
+ * REST controller for managing bonus
+ */
+@RestController
+@RequestMapping("/api")
+public class BonusResource {
+
+    private final Logger log = LoggerFactory.getLogger(BonusResource.class);
+
+    private static final String ENTITY_NAME = "bonus";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final BonusService bonusService;
+
+    private final BonusRepository bonusRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final GradingScaleRepository gradingScaleRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public BonusResource(BonusService bonusService, BonusRepository bonusRepository, CourseRepository courseRepository, GradingScaleRepository gradingScaleRepository,
+            AuthorizationCheckService authCheckService) {
+        this.bonusService = bonusService;
+        this.bonusRepository = bonusRepository;
+        this.courseRepository = courseRepository;
+        this.gradingScaleRepository = gradingScaleRepository;
+        this.authCheckService = authCheckService;
+    }
+
+    /**
+     * GET /courses/{courseId}/exams/{examId}/bonus : Find bonus model for exam (where Bonus.bonusToGradingScale corresponds to the exam)
+     * Sets Bonus.bonusStrategy from the bonus strategy set on the exam's grading scale.
+     *
+     * @param courseId the course to which the exam belongs
+     * @param examId   the exam to which the bonus belongs
+     * @return ResponseEntity with status 200 (Ok) with body the bonus source if it exists and 404 (Not found) otherwise
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/bonus")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<Bonus> getBonusForExam(@PathVariable Long courseId, @PathVariable Long examId) {
+        log.debug("REST request to get bonus sources for exam: {}", examId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        var bonus = bonusRepository.findAllByBonusToExamId(examId).stream().findAny().orElseThrow(() -> new EntityNotFoundException("BonusToGradingScale exam", examId));
+        bonus.setBonusStrategy(bonus.getBonusToGradingScale().getBonusStrategy());
+        filterBonusForResponse(bonus);
+        return ResponseEntity.ok(bonus);
+    }
+
+    /**
+     * GET bonus/:bonusId : get the bonus with id.
+     * Sets Bonus.bonusStrategy from the bonus strategy set on the exam's grading scale.
+     *
+     * @param bonusId the id of the bonus to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the bonus, or with status 404 (Not Found)
+     */
+    @GetMapping("bonus/{bonusId}")
```

Comment:
wrong URL

Reply:
Url is standardized by adding `courseId` and `examId`.